src/types/index.ts

```diff
@@ -1,46 +1,392 @@
-// @ts-ignore
-interface AutoComplete {}
+// -----------------------------------------------------------------------------
+//  --------------------------------- TYPES ------------------------------------
+// -----------------------------------------------------------------------------
 
-interface StopGroup {}
+type GroundPlace = StopGroup | StopCluster;
 
-interface StopCluster {}
+type GroundPlaces = Record<Gpuid, GroundPlace>;
 
-interface StopGroupProperties {}
+type Gpuid = string;
 
-interface StopClusterProperties {}
+type StopGroupGpuid = Gpuid;
 
-interface Filters {}
+type StopClusterGpuid = Gpuid;
 
-interface GroundPlacesDiff {}
+// -----------------------------------------------------------------------------
+//  ------------------------------- INTERFACES ---------------------------------
+// -----------------------------------------------------------------------------
 
-type Gpuid = string;
+interface StopGroup {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.GROUP;
+  childs: SegmentProviderStop[];
+  serviced: Serviced;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type StopGroupGpuid = Gpuid;
+interface StopCluster {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.CLUSTER;
+  childs: StopGroupGpuid[];
+  serviced: Serviced;
+  unique_name?: string;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type StopClusterGpuid = Gpuid;
+interface SegmentProviderStop {
+  unique_name: null;
+  company_name: string;
+  name: string;
+  latitude: number;
+  serviced: Serviced;
+  company_id: number;
+  longitude: number;
+  id: string;
+}
+
+interface CreateStopGroupProperties {
+  countryCode: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType;
+  serviced: Serviced;
+}
+
+interface CreateStopClusterProperties {
+  countryCode: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType;
+  createdFromStopGroupGpuid: StopGroupGpuid;
+  serviced: Serviced;
+}
+
+interface UpdateStopProperties {
+  name?: string;
+  longitude?: number;
+  latitude?: number;
+}
+
+interface GroundPlacesDiff {
+  [gpuid: string]: {
+    type: GroundPlaceActionType;
+    into?: string;
+  };
+}
+
+// TODO: Implement these interfaces
+
+interface AutoComplete {}
+
+interface Filters {}
+
+// -----------------------------------------------------------------------------
+//  --------------------------------- ENUMS ------------------------------------
+// -----------------------------------------------------------------------------
+
+enum GroundPlaceActionType {
```

Comment:
prefix with E for ENUM

Reply:
It's a question of programming style but I'm not really a fan of adding a prefix in front of its interfaces / types, it's a practice that comes from a time when IDEs were not as developed as they are now.

`Interfaces`, `types` and `enums` are the only ones that are in PascalCase while variables are in camelCase, so it's pretty obvious from the code review.

If I had to add E before each enum here, I would also have to do it for each interface with an I and for each type with a T in front of it. I don't think that's of any help, but if you all agree on this convention, then I'll go to it.

Reply:
@rimiti & @Theo-Le-Donne will choose.

Reply:
Setting a prefix is ​​an our convention, but you are not mandatory to do.

Reply:
I find it more readable with prefix, but do it without if thats what you're used to

Reply:
Ok thanks both, I'll stay for now with "no prefix", this should be ok :)

Reply:
I prefer the prefix also. But as you want.